File: leetcode/2540.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var getCommon = function(nums1, nums2) {
  const set = new Set(nums1);

  for (const n2 of nums2) {
    if (set.has(n2)) {
      return n2;
    }
  }

  return -1;
};

File: leetcode/2540.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        if len(nums1) > len(nums2):
            search, base = nums2, nums1
        else:
            search, base = nums1, nums2

        for n in search:
            if n >= base[0] and n in base:
                return n
                
        return -1
